### DISCUSSION TITLE
What 3rd party LUA libraries/modules are included?

### DISCUSSION THREAD

**Discussion by Jakobud** [mamedev/discussions#162] [Q&A]
I noticed that some plugins use a `require('json')` module in their code. Is this a 3rd party LUA module/library that is bundled in the LUA API? Something like https://github.com/rxi/json.lua ?

Are there other bundled libraries that are included in the API? Is there a documented list of them so that I know what is available?

**Comment by Jakobud** [comment]
Or is the `require('json')` referring to requiring the included JSON plugin into your own plugin?

**Comment by cuavas** [comment]
MAME includes zlib, lfs, linenoise and sqlite3 Lua extensions.

The JSON script is luv-json, as you can see in the plugins folder: https://github.com/mamedev/mame/tree/master/plugins/json

**Reply by Jakobud** [reply]
Okay so when you do `require` is it trying to require something out of the plugin folder?
